src/transformers/models/whisper/generation_whisper.py

```diff
@@ -474,6 +474,15 @@ def generate(
                 "The input name `inputs` is deprecated. Please make sure to use `input_features` instead.",
                 FutureWarning,
             )
+
+        if "assistant_model" in kwargs:
```

Comment:
This check should live in `generate`, as it's relevant to all encoder-decoder models. Furthermore, it should be checking that:
1. If the main model is `AutoModelForSpeechSeq2Seq` (i.e. encoder-decoder) and the assistant model is `AutoModelForCausalLM` (i.e. decoder-only), then the encoder dims have to match (as well as the tokeniser's). Ideally, we'd check for [all `encoder_xyz` attributes](https://huggingface.co/openai/whisper-large-v3/blob/1ecca609f9a5ae2cd97a576a9725bc714c022a93/config.json#L23-L26), not just `encoder_attention_heads`
2. If the main model is `AutoModelForSpeechSeq2Seq` (i.e. encoder-decoder) and the assistant model is also `AutoModelForSpeechSeq2Seq` (i.e. encoder-decoder), then the encoder dims can be different. The only criteria here is that both model use the same tokenizer (we can do a shallow check by looking at the vocab size and making sure they're the same)

Reply:
Would be better, but as far as I know, those are the conditions for audio models no ? shouldn't we also generalize these observations to non-audio models compatible with assistants ?